src/main/java/com/book/service/UserService.java

```diff
@@ -0,0 +1,35 @@
+package com.book.service;
+
+import com.book.model.UsersEntity;
+import com.book.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class UserService implements UserServiceInterface{
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(UserRepository userRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    @Transactional
+    public UsersEntity registerUser(UsersEntity usersEntity) {
+        usersEntity.setUserType("user");
+
+        if(userRepository.existsByUserName(usersEntity.getUserName())){
+            return null;
```

Comment:
try to avoid returning null. Instead, try using java 8 Optional: https://www.baeldung.com/java-optional